Makefile.am

```diff
@@ -71,3 +73,16 @@ install-library:
 
 install-headers:
 	cd src && $(MAKE) $@
+
+compile-commands:
+if HAVE_CLANG
+	$(MAKE) -f src/Makefile clean
+	TMP_DIR=$$(mktemp -d --suffix=.suricata-compile-commands) && \
+	$(MAKE) EXTRA_CFLAGS="-MJ $$TMP_DIR/\$$*.json" && \
+	echo '[' > $(top_builddir)/compile_commands.json && \
+	cat $$TMP_DIR/*.json >> $(top_builddir)/compile_commands.json && \
+	echo ']' >> $(top_builddir)/compile_commands.json && \
+	rm -rf $$TMP_DIR
+else
+	@echo "clang compile_commands.json generation not supported with current compiler"
+endif
```

Comment:
Back-to-back invocations of this result in an error, as no files are rebuilt. Is that intended?

Should the first line be: `$(MAKE) -C src clean` or some other invocation that actually runs `clean` in the `src` directory?

Reply:
Outch, sorry I forgot to test that one. I think I was so happy to get the file generated I forgot about this test.

Reply:
Pushed the proposed fix.